keras_hub/src/models/basnet/basnet.py

```diff
@@ -0,0 +1,128 @@
+import keras
+
+from keras_hub.src.api_export import keras_hub_export
+from keras_hub.src.models.basnet.basnet_backbone import BASNetBackbone
+from keras_hub.src.models.basnet.basnet_preprocessor import BASNetPreprocessor
+from keras_hub.src.models.image_segmenter import ImageSegmenter
+
+
+@keras_hub_export("keras_hub.models.BASNetImageSegmenter")
+class BASNetImageSegmenter(ImageSegmenter):
+    """
+    A Keras model implementing the BASNet architecture for semantic
+    segmentation.
+
+    References:
+    [BASNet: Boundary-Aware Segmentation Network for Mobile and Web Applications](https://arxiv.org/abs/2101.04704)
+
+    Args:
+        backbone: A `keras_hub.models.BASNetBackbone` instance.
+        preprocessor: `None`, a `keras_hub.models.Preprocessor` instance,
+            a `keras.Layer` instance, or a callable. If `None` no preprocessing
+            will be applied to the inputs.
+
+    Example:
+    ```python
+    import keras_hub
+
+    images = np.ones(shape=(1, 288, 288, 3))
+    labels = np.zeros(shape=(1, 288, 288, 1))
+
+    image_encoder = keras_hub.models.ResNetBackbone.from_preset(
+        "resnet_18_imagenet",
+        load_weights=False
+    )
+    backbone = keras_hub.models.BASNetBackbone(
+        image_encoder,
+        num_classes=1,
+        image_shape=[288, 288, 3]
+    )
+    model = keras_hub.models.BASNetImageSegmenter(backbone)
+
+    # Evaluate the model
+    pred_labels = model(images)
+
+    # Train the model
+    model.compile(
+        optimizer="adam",
+        loss=keras.losses.BinaryCrossentropy(from_logits=False),
+        metrics=["accuracy"],
+    )
+    model.fit(images, labels, epochs=3)
+    ```
+    """
+
+    backbone_cls = BASNetBackbone
+    preprocessor_cls = BASNetPreprocessor
+
+    def __init__(
+        self,
+        backbone,
+        preprocessor=None,
+        **kwargs,
+    ):
+
+        # === Functional Model ===
+        x = backbone.input
+        outputs = backbone(x)
+        # only return the refinement module's output as final prediction
+        outputs = outputs["refine_out"]
+        super().__init__(inputs=x, outputs=outputs, **kwargs)
+
+        # === Config ===
+        self.backbone = backbone
+        self.preprocessor = preprocessor
+
+    def compute_loss(self, x, y, y_pred, *args, **kwargs):
+        # train BASNet's prediction and refinement module outputs against the same ground truth data
```

Comment:
wrap at 80 characters. recent formatting changes will catch this if you sync with master